client/src/components/ProfileListings/ProfileCard/ProfileCard.tsx

```diff
@@ -0,0 +1,66 @@
+import { useState } from 'react';
+import { useHistory } from 'react-router';
+import { Box, Paper, Typography } from '@material-ui/core';
+import Location from '@material-ui/icons/LocationOn';
+import Rating from '@material-ui/lab/Rating';
+import { Profile } from '../../../interface/Profile';
+import useStyles from './useStyles';
+
+interface Props {
+  profile: Profile;
+}
+export default function ProfileCard({ profile }: Props): JSX.Element {
+  const classes = useStyles();
+  const [isMouseOver, setIsMouseOver] = useState(false);
+  const history = useHistory();
+
+  const { address, _id: id, profilePic, introduction, pitch, rate, firstName, lastName } = profile;
+
+  const handleCardSelect = () => history.push(`/profile/${id}`);
```

Comment:
If you implement the `CardActionArea` you can use react-router's `Link` to handle routing (generally it is a good practice to stay away from manipulating the history stack and let react-router handle the routing)

Eg: 
```
<CardActionArea component={Link} to={'/profile'}>
    // ...
</CardActionArea >
```